File: Creational Patterns/Singleton/Singleton.swift

class Singleton1 {
    
    private init() {}
    
    private static var uniqueValue: Singleton1?
    
    public func get_instance() -> Singleton1 {
        if Singleton1.uniqueValue == nil {
            Singleton1.uniqueValue = Singleton1()
        }
        return Singleton1.uniqueValue!
    }
}

class Singleton {
    
    private init() {}
    
    private static let uniqueValue = Singleton()
}

File: Creational Patterns/Singleton/CS/Singleton.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Singleton
{
    class Singleton
    {
        private static Singleton instance;
        public string namePattern { get; private set; }
        protected Singleton(string name) 
        {
            this.namePattern = name;
        }

        public static Singleton getInstance(string name)
        {
            if (instance == null)
            {
                instance = new Singleton(name);
            }

            return instance;
        }
    }
}
